TLM/TLM/Patch/_VehicleAI/_CargoTruckAI/StartPathFindPatch.cs

```diff
@@ -0,0 +1,32 @@
+namespace TrafficManager.Patch._VehicleAI._CargoTruckAI{
+    using HarmonyLib;
+    using JetBrains.Annotations;
+    using TrafficManager.Patch._PathManager;
+    using System.Reflection;
+    using TrafficManager.API.Traffic.Enums;
+    using TrafficManager.Manager.Impl;
+    using System.Reflection.Emit;
+    using System.Collections.Generic;
+
+
+    [HarmonyPatch]
+    public class StartPathFindPatch {
+
+        [UsedImplicitly]
+        public static MethodBase TargetMethod() => StartPathFindCommons.TargetMethod<CargoTruckAI>();
+
+        [UsedImplicitly]
+        public static void Prefix(ushort vehicleID, ref Vehicle vehicleData) {
+            ExtVehicleManager.Instance.OnStartPathFind(vehicleID, ref vehicleData, null);
+            CreatePathPatch.ExtVehicleType = ExtVehicleType.CargoVehicle;
+            CreatePathPatch.ExtPathType = ExtPathType.None;
+            CreatePathPatch.VehicleID = vehicleID;
+        }
+
+        [UsedImplicitly]
+        [HarmonyPriority(Priority.Low)] // so that if this code is redundant, it would result in warning log.
```

Comment:
Would it be worth moving these transpilers in to separate file - ie. collate all the `81 Tiles` compatibilty patches in one place? We only need to do those transpiles if 81 Tiles mod is active?

Reply:
Also, is there a way to conditionally patch? If not it would be useful addition to Harmony - "only do this patch if this condition is true".

Reply:
Its too much trouble and sophistication. It also can go wrong in different ways.
In any case, this would be out of scope. The objective of this PR is to copy Redirection code to Harmony, without any change to functionality. Maybe fix some simple bugs and add some comments.
I think 81 tile mod is the right place to have these transpilers so probably not worth the effort to do anything fancy here if we are going to remove it anyway.

> Would it be worth moving these transpilers in to separate file 

Its already in a seperate file in `StartPathFindsCommons()`

> is there a way to conditionally patch

Yes, the transpiler is a C# code so it could check if 81 tile does not exist then don't do anything.

I created issue #953 as follow up.